models/src/main/java/theory/BooleanAlgebraSubstAndOutputs.java

```diff
@@ -11,16 +11,24 @@
 /**
  * BooleanAlgebraSubst: A Boolean Alegbra with substitution
  * @param <P> The type of predicates forming the Boolean algebra
+ * @param <PL> The type of predicates over lists of variables 
  * @param <F> The type of functions S->S in the Boolean Algebra 
  * @param <S> The domain of the Boolean algebra
  */
 public abstract class BooleanAlgebraSubstAndOutputs<P, PL, F, S> extends BooleanAlgebraSubst<P, F, S>{
 	
 	/**
-	 * hacky for now
+	 * Returns a list of P predicates whose conjunction expresses the PL p
```

Comment:
Can you add a bit about wht the list of predicates represent (same comment for monadic)

e.g., [x=1,x=1] stands for phi(x,y)=x=1 /\ y=1

Reply:
Added in new commit